### DISCUSSION TITLE
Path to assets

### DISCUSSION THREAD

**Discussion by antoinelacroix68** [requarks/wiki#3758] [Help / Questions] [answered]
Hello, I don't know if what I'm asking makes sense but where are all the assets we upload on the wiki stored ? 

For context, my wiki js is running on a linux container.

I'm asking that because I'm basically migrating an old dokuwiki to wiki js and i would want to store all assets from this one (images and videos mostly) into the new wiki (wiki js) quickly... Thanks in advance

**Comment by adenoz** [comment]
I'm no expert but I believe by default they are stored in the root directory. However you should probably create a sud-directory like 'media' to keep things more organised. I am not familiar with migrating but hope this helps until someone with deeper experience comes along.

**Comment by NGPixel** [comment, marked as answer]
Assets are stored in the DB. Frequently accessed assets are also stored on disk as cache (usually under /wiki/data/cache).

**Reply by antoinelacroix68** [reply]
Ok thanks alot for your quick answer !

**Reply by antoinelacroix68** [reply]
I just wanna be sure about something, I tried uploading videos on the database but the upload systematically fails. i assume that's because the DB doesn't support videos so it means I'd have to host these elsewhere the put an iframe on mi wikijs pages, am I right ?

**Reply by NGPixel** [reply]
You shouldn't upload videos to your wiki. This isn't the right place to host them and you're most likely to hit the file size limit. Host them either on a file service like AWS S3 / Azure Blob Storage or on a video service like Youtube and embed them.

**Reply by antoinelacroix68** [reply]
Ok that's what I thought, thanks for your answers you just earned me a day of work for something that wouldn't work anyway.

**Reply by wzrdtales** [reply]
@NGPixel is support for uploading them to a s3 backend or similar stateless storage planned (so as a gateway through wiki js)? Saving it to the db doesn't sound that good to me, especially considering backups.

**Reply by NGPixel** [reply]
Yes, coming in v3

**Reply by wzrdtales** [reply]
:+1: looking forward to it. Feel free to reach out to me btw. currently due to lack of time I through my company majorly only contribute by money, but I may help out in the future in other ways as well.
